parlai/core/metrics.py

```diff
@@ -63,6 +64,70 @@ def _score(g_tokens, a_tokens):
     return max(scores)
 
 
+def aggregate_metrics(reporters):
+    #reporters is a list of teachers or worlds
+    m = {}
+    m['tasks'] = {}
+    sum_accuracy = 0
+    sum_f1 = 0
+    num_tasks = 0
+    total = 0
+    for i in range(len(reporters)):
+        tid = reporters[i].getID()
+        mt = reporters[i].report()
+        while tid in m['tasks']:
+            # prevent name cloberring if using multiple tasks with same ID
+            tid += '_'
+        m['tasks'][tid] = mt
+        total += mt['total']
+        if 'accuracy' in mt:
+            sum_accuracy += mt['accuracy']
+            num_tasks += 1
+            if 'f1' in mt:
+                sum_f1 += mt['f1']
+    m['total'] = total
+    m['accuracy'] = 0
+    if num_tasks > 0:
+        m['accuracy'] = sum_accuracy / num_tasks
+        if sum_f1 > 0:
+            m['f1'] = sum_f1 / num_tasks
+    return m
+
+
+def compute_time_metrics(world, opt):
```

Comment:
this isnt really to do with the metrics class but i see why you've put it here.

it has an unhealthy use of options that only depend on train_model main (max_train_time), and requires the world too..

but i dont see how to do it better right now, other than making those all arguments instead?